Source/WebCore/workers/service/context/ServiceWorkerInspectorProxy.cpp

```diff
@@ -59,14 +59,17 @@ void ServiceWorkerInspectorProxy::serviceWorkerTerminated()
     m_channel = nullptr;
 }
 
-void ServiceWorkerInspectorProxy::connectToWorker(FrontendChannel& channel)
+void ServiceWorkerInspectorProxy::connectToWorker(FrontendChannel& channel, bool isAutomaticConnection, bool immediatelyPause)
 {
     m_channel = &channel;
 
     RefPtr serviceWorkerThreadProxy = m_serviceWorkerThreadProxy.get();
     SWContextManager::singleton().setAsInspected(serviceWorkerThreadProxy->identifier(), true);
-    serviceWorkerThreadProxy->thread().runLoop().postDebuggerTask([] (ScriptExecutionContext& context) {
-        downcast<WorkerGlobalScope>(context).inspectorController().connectFrontend();
+    serviceWorkerThreadProxy->thread().runLoop().postDebuggerTask([isAutomaticConnection, immediatelyPause, serviceWorkerThreadProxy](ScriptExecutionContext& context) {
+        Function<void()> handleFrontendInitialized = [serviceWorkerThreadProxy] {
+            serviceWorkerThreadProxy->serviceWorkerDebuggable().unpauseForInitializedInspector();
+        };
```

Comment:
is there really no way to get to the `ServiceWorkerDebuggable` from within `WorkerInspectorController`?  it seems really overkill (and annoying for object lifetime tracking) to have to pass in a lambda

NIT: id just inline this

Reply:
I simplified this a bit.

Inlining could also work but I thought this makes it more compact that only the callback uses the debuggable, rather than having the controller hold a (weak) reference to it that's only used one time.

There seems to not be an easy way to get the debuggable from within the controller, and the controller is not specialized for service workers so maybe it makes less sense to do service-worker-specific work in there.